DOI.js

```diff
@@ -93,25 +96,28 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(DOIs, doc) {
+function retrieveDOIs(DOIs, doc, provider) {
```

Comment:
I would make `provider` into `providers` and have that be an array containing translator IDs that should be used for DOIs. On each iteration of `retrieveDOIs` you would `.shift` the translator ID to use in `.setTranslator` until you run out of IDs. IMO that would make this simpler.